### PR TITLE
[chore] Use consistent go-github versioning

### PR DESCRIPTION
**Pull request by pjlast**
We have accumulated an icky amount of different go-github versions. This PR cleans it up a bit

There are 2 cases that still use v48 instead of v55, this is because the `dghubble/gologin` package we use has a hard dependency on it. A future PR might abstract this a bit.

## Test plan

Trusting the system

### CODE REVIEW

**Review by unknwon — APPROVED** [review]
PR intention LGTM

File: go.mod [hunk 1]
```diff
@@ -49,7 +49,6 @@ replace (
 // =================================
 // These entries indicate replace directives that are defined for unknown reasons.
 replace (
-	github.com/dghubble/gologin => github.com/sourcegraph/gologin v1.0.2-0.20181110030308-c6f1b62954d8
```
Comment (eseliger) [inline go.mod:52]: so do we know what the "unknown reason" we overwrite this was?

Comment (pjlast) [reply go.mod:52]: lmao no, but I updated this as well so 🤷‍♂️

**Review by eseliger — APPROVED** [review]
pending inline comment

File: go.mod (hunk 1 above)
Comment (eseliger) [reply go.mod:52]: I just wonder if there's some critical fix in there that would get lost now .. :scary: 
Also, we can probably archive that fork after this is merged :D

Comment (eseliger) [reply go.mod:52]: this is the diff: https://github.com/dghubble/gologin/compare/main...sourcegraph:gologin:master says it adds support for github enterprise(?)

Comment (pjlast) [reply go.mod:52]: Okay judging by the commits in https://github.com/sourcegraph/gologin the fork was to add support for enterprise clients, but this is supported on its own now

Comment (pjlast) [reply go.mod:52]: (infact the NewEnterpriseClient that the fork is using is deprecated now)

Comment (eseliger) [reply go.mod:52]: cool, thanks for double checking

**Pull request merged**